compliance/collection/nodeinventorizer/caching_inventorizer_test.go

```diff
@@ -206,18 +220,49 @@ func (s *TestComplianceCachingSuite) TestScanReturnsCachedInventory() {
 	maxBackoff := 10 * time.Second
 	inventoryCachePath := fmt.Sprintf("%s/inventory-cache", s.T().TempDir())
 	cs := *NewCachingScanner(mockScanner{}, inventoryCachePath, cache, initial, maxBackoff, func(time.Duration) {})
+	validUntil := time.Now().Add(2 * time.Minute)
 
 	w := &inventoryWrap{
-		CacheValidUntil:      time.Now().Add(2 * time.Minute),
+		CacheValidUntil:      validUntil,
 		RetryBackoffDuration: 0,
 		CachedInventory:      "{\n  \"nodeId\": \"00000000-0000-0000-0000-000000000000\",\n  \"nodeName\": \"cachedNode\",\n  \"scanTime\": \"2023-11-11T11:11:11.382478080Z\",\n  \"components\": {\n    \"namespace\": \"unknown\"\n  },\n  \"notes\": [\n    \"LANGUAGE_CVES_UNAVAILABLE\"\n  ]\n}",
 	}
 	s.writeWrap(w, inventoryCachePath)
 
+	// Check directly returned results
 	actual, err := cs.Scan("testme")
-
 	s.NoError(err)
 	s.Equal(actual.NodeName, "cachedNode")
+
+	// Check cached results
+	wrap := s.readWrap(inventoryCachePath)
+	s.WithinDuration(validUntil, wrap.CacheValidUntil, 100*time.Millisecond, "Cache validity should not change when accessing a cached scan.")
+}
+
+func (s *TestComplianceCachingSuite) TestScanRunsNewInventory() {
+	initial := 2 * time.Second
+	cache := initial
+	maxBackoff := 10 * time.Second
+	inventoryCachePath := fmt.Sprintf("%s/inventory-cache", s.T().TempDir())
+	cs := *NewCachingScanner(mockScanner{}, inventoryCachePath, cache, initial, maxBackoff, func(time.Duration) {})
+	validUntil := time.Now().Add(-1 * time.Minute)
+
+	w := &inventoryWrap{
+		CacheValidUntil:      validUntil,
+		RetryBackoffDuration: 0,
+		CachedInventory:      "{\n  \"nodeId\": \"00000000-0000-0000-0000-000000000000\",\n  \"nodeName\": \"cachedNode\",\n  \"scanTime\": \"2023-11-11T11:11:11.382478080Z\",\n  \"components\": {\n    \"namespace\": \"unknown\"\n  },\n  \"notes\": [\n    \"LANGUAGE_CVES_UNAVAILABLE\"\n  ]\n}",
+	}
+	s.writeWrap(w, inventoryCachePath)
+
+	// Check directly returned results
+	actual, err := cs.Scan("testme")
+	s.NoError(err)
+	s.Equal(actual.NodeName, "testme")
+
+	// Check cached results
+	wrap := s.readWrap(inventoryCachePath)
```

Comment:
A check that `wrap.CacheValidUntil` is in the future will make the test more explicit. Please add one.

Reply:
Done